### PR TITLE
fix(eventbus): allow clients with istio sidecar. Fixes: #1311

### PR DESCRIPTION
**Pull request by antoniomo**
Istio-related issues affecting this behaviour:

- https://github.com/nats-io/nats-operator/issues/88
- https://github.com/istio/istio/issues/28623

The fix is very simple. I added the related issues to the code comment,
let me know if that's undesirable or there's something else to clarify.

Checklist:

* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo-events/blob/master/USERS.md).

### CODE REVIEW

**Comment by antoniomo** [comment]
Fixes: #1311

File: controllers/eventbus/installer/nats.go [hunk 1]
```diff
@@ -414,7 +414,11 @@ func (i *natsInstaller) buildStanService() (*corev1.Service, error) {
 		Spec: corev1.ServiceSpec{
 			ClusterIP: corev1.ClusterIPNone,
 			Ports: []corev1.ServicePort{
-				{Name: "client", Port: clientPort},
+				// Prefix tcp- to enable clients to connect from
+				// an istio-enabled namespace, following:
+				// https://github.com/nats-io/nats-operator/issues/88
+				// https://github.com/istio/istio/issues/28623
+				{Name: "tcp-client", Port: clientPort},
```
Comment (whynowy) [inline controllers/eventbus/installer/nats.go:421]: Have you tested if this will impact upgrading from an existing EventBus?

Comment (antoniomo) [reply controllers/eventbus/installer/nats.go:421]: Hi!

Good point, I need to test with and without Istio (not just with). I will have time in the weekend to do more thoroughly testing, any particular scenario you would like to see checked?

- Without Istio, run pre-patch, apply post-patch, everything should still work.
- With Istio, run pre-patch (doesn't work), apply post-patch, everything should work.
- Fresh installs on both cases.

Does this cover it?

Comment (whynowy) [reply controllers/eventbus/installer/nats.go:421]: Sorry for the late response, just came back from vacation.

I'm more concerned about if the port name change is allowed on an existing service.

Comment (antoniomo) [reply controllers/eventbus/installer/nats.go:421]: Got it! I'll check that as well, some delays getting time on my side.

Comment (whynowy) [reply controllers/eventbus/installer/nats.go:421]: I tested it, upgrading has no issue, thanks!

**Review by whynowy — APPROVED** [review]

File: controllers/eventbus/installer/nats.go (hunk 1 above)
Comment (antoniomo) [reply controllers/eventbus/installer/nats.go:421]: Awesome! Thank you very much! I was unable to get time for this, deployments "season" at work have kept me busy and I kept postponing it :/

Thanks again!

**Pull request merged**